### PR TITLE
[Tests-only] Run `append scenario title to serverlog` code  only in CI

### PR DESCRIPTION
**Pull request by amrita-shrestha**
### Description
This PR 
- refactor appending scenario title to serverlog.log  so that it runs only in CI 

### Related Issue:
- https://github.com/owncloud/client/issues/9826

### CODE REVIEW

File: test/gui/shared/scripts/bdd_hooks.py [hunk 1]
```diff
@@ -127,10 +128,14 @@ def hook(context):
         filename = context._data["title"].replace(" ", "_") + ".png"
         directory = os.environ["GUI_TEST_REPORT_DIR"] + "/screenshots"
 
-        if not os.path.exists(directory):
+        while not (os.path.exists(directory)):
             os.makedirs(directory)
```
Comment (saw-jan) [inline test/gui/shared/scripts/bdd_hooks.py:132]: You might get infinite loop (in rare case). `if` is good to go.
```suggestion
        if not os.path.exists(directory):
            os.makedirs(directory)
```

**Review by sushmita56 — COMMENTED** [review]
run `black test/gui` command and push again

File: test/gui/shared/scripts/bdd_hooks.py [hunk 2]
```diff
@@ -127,10 +128,14 @@ def hook(context):
...
 
-        pb.savev(os.path.join(directory, filename), "png", [], [])
+        try:
+            pb.savev(os.path.join(directory, filename), "png", [], [])
+        except:
+            pb.savev(os.path.join(directory, filename), "png", [], [])
```
Comment (saw-jan) [inline test/gui/shared/scripts/bdd_hooks.py:137]: ignore the screenshot file if it doesn't exists.
```suggestion
        try:
            pb.savev(os.path.join(directory, filename), "png", [], [])
        except:
            pass
```

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by sushmita56 — COMMENTED** [review]
LGTM

File: test/gui/shared/scripts/bdd_hooks.py [hunk 3]
```diff
@@ -53,15 +53,16 @@ def hook(context):
...
-    guiTestReportDir = os.environ.get("GUI_TEST_REPORT_DIR")
-    f = open(guiTestReportDir + "/serverlog.log", "a")
-    f.write(
-        str((datetime.now()).strftime("%H:%M:%S:%f"))
-        + "\tBDD Scenario: "
-        + context._data["title"]
-        + "\n"
-    )
-    f.close()
+    if os.getenv('CI'):
```
Comment (individual-it) [inline test/gui/shared/scripts/bdd_hooks.py:56]: don't we need the server log when running the tests locally?

File: test/gui/shared/scripts/bdd_hooks.py [hunk 4]
```diff
@@ -130,7 +131,10 @@ def hook(context):
         if not os.path.exists(directory):
             os.makedirs(directory)
 
-        pb.savev(os.path.join(directory, filename), "png", [], [])
+        try:
+            pb.savev(os.path.join(directory, filename), "png", [], [])
+        except:
+            pass
```
Comment (individual-it) [inline test/gui/shared/scripts/bdd_hooks.py:137]: why is that upload failing, and if it happens the developer will be puzzled by not seeing the screenshot.
I think we should not just cover the issue but at lease report that the upload failed

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/shared/scripts/bdd_hooks.py (hunk 4 above)
Comment (saw-jan) [inline test/gui/shared/scripts/bdd_hooks.py:137]: @amrita-shrestha 
There seems to be another issue with the screenshot. So I would recommend not to include fixes for screenshot here but instead create a new issue and investigate the issue and make fix in another PR.

**Review by saw-jan — CHANGES_REQUESTED** [review]
@sushmita56 Please, link the issue in this comment as well

File: test/gui/shared/scripts/bdd_hooks.py (hunk 4 above)
Comment (amrita-shrestha) [reply test/gui/shared/scripts/bdd_hooks.py:137]: ok

File: test/gui/shared/scripts/bdd_hooks.py (hunk 3 above)
Comment (amrita-shrestha) [reply test/gui/shared/scripts/bdd_hooks.py:56]: Locally `server log` can be viewed in the terminal. In my view, we don't need this while running tests locally.

File: test/gui/shared/scripts/bdd_hooks.py (hunk 4 above)
Comment (amrita-shrestha) [reply test/gui/shared/scripts/bdd_hooks.py:137]: For the screenshot, I will create the next PR.

**Review by individual-it — APPROVED** [review]

**Review by saw-jan — APPROVED** [review]
LGTM
